client/src/components/ProfileSettings/EditProfile/EditProfileForm/EditProfileForm.tsx

```diff
@@ -0,0 +1,353 @@
+import DateFnsUtils from '@date-io/date-fns';
+import { Grid, Switch, Typography } from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { Formik, FormikHelpers } from 'formik';
+import { useState } from 'react';
+import * as Yup from 'yup';
+import { Profile, ProfileType } from './../../../../interface/Profile';
+import AvailableSwitch from './AvailableSwitch/AvailableSwitch';
+import useStyles from './useStyles';
+
+interface Props {
+  handleSubmit: (
+    {
+      type,
+      firstName,
+      lastName,
+      gender,
+      dateOfBirth,
+      email,
+      phoneNumber,
+      address,
+      description,
+      isAvailable,
+      availableHoursPerWeek,
+      hourlyRate,
+    }: Profile,
+    { setStatus, setSubmitting }: FormikHelpers<Profile>,
+  ) => void;
+}
+
+const EditProfileForm = ({ handleSubmit }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  const [isAvailable, setIsAvailable] = useState(true);
+  const handleToggle = () => {
+    isAvailable ? setIsAvailable(false) : setIsAvailable(true);
+  };
```

Comment:
Do we need this state, is there a problem using `Formik`'s state for these instead?

Reply:
@jedchernandez